core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -470,12 +470,18 @@ public static long mapRO(FilesFacade ff, long fd, long size, int memoryTag) {
         return mapRO(ff, fd, size, 0, memoryTag);
     }
 
+    /**
+     * Maps a file in read-only mode.
+     *
+     * Important note. Linux requires the offset to be page aligned.
+     */
     public static long mapRO(FilesFacade ff, long fd, long size, long offset, int memoryTag) {
         final long address = ff.mmap(fd, size, offset, Files.MAP_RO, memoryTag);
```

Comment:
We can put assert here so that all platforms will check that offset is page aligned

Reply:
Added assert in 9678f2dec1966a917f63dd78de06e01c1d2d0740